src/pages/HomePage/ui/HomePage.tsx

```diff
@@ -0,0 +1,111 @@
+import { useEffect, useState } from 'react';
+import { SearchPage } from '@pages/SearchPage';
+import { Header } from '@widgets/header';
+import { ErrorBoundary } from '@shared/ui/ErrorBoundary';
+import { fetchItems, Item, Pagination as PaginationInterface } from '@shared/api/items';
+
+import '@shared/styles/global.css';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+export const HomePage = () => {
+  const [userValue, setUserValue] = useState('');
+  const [inputValue, setInputValue] = useState('');
+  const [items, setItems] = useState<Item[]>([]);
+  const [itemsPerPage, setItemsPerPage] = useState(10);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [pagination, setPagination] = useState<PaginationInterface>({
+    current_page: 1,
+    has_next_page: false,
+    items: {
+      count: 1,
+      total: 1,
+      per_page: 10,
+    },
+    last_visible_page: 1,
+  });
+  const [isFetching, setIsFetching] = useState(false);
+  const [isResultEmpty, setIsResultEmpty] = useState(false);
+
+  const [searchParams] = useSearchParams({
+    q: '',
+    limit: '10',
+    page: '1',
+  });
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const newQuery = searchParams.get('q') || '';
+    const newLimit = parseInt(searchParams.get('limit') || '10', 10);
+    const newPage = parseInt(searchParams.get('page') || '1', 10);
+    setUserValue(newQuery);
+    setItemsPerPage(newLimit);
+    setCurrentPage(newPage);
+    fetchByName();
+  }, [searchParams]);
+
+  const handleInputName = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setUserValue(event.target.value.trimStart().trimEnd());
+    setInputValue(event.target.value);
+  };
+  const prevPageHandler = () => {
+    setCurrentPage((prev) => prev - 1);
+    navigate(`?q=${userValue}&limit=${itemsPerPage}&page=${currentPage - 1}`);
+  };
+  const nextPageHandler = () => {
+    setCurrentPage((prev) => prev + 1);
+    navigate(`?q=${userValue}&limit=${itemsPerPage}&page=${currentPage + 1}`);
```

Comment:
maybe `useRouter()` and `router.push(
...,
query: {
  limit, page,  ...
}
)` would be better?
and try to get rid off copy-pasting here, navigate lines looks really similar